src/main/kotlin/org/rust/lang/core/types/infer/MemoryCategorization.kt

```diff
@@ -123,14 +126,22 @@ private fun processDeref(baseCmt: Cmt): Cmt {
     }
 }
 
+private fun processRvalue(expr: RsExpr): Cmt =
+    Cmt(expr.type, MutabilityCategory.Declared)
+
+private fun isRvalue(expr: RsExpr): Boolean =
```

Comment:
Maybe create some marker interface for this expression types to use `expr is RsSomeInterfaceName` instead of this function?

Reply:
Actually this code is redundant.